### DISCUSSION TITLE
Avoid adding temporary files/only add modified ones

### DISCUSSION THREAD

**Discussion by janopae** [stefanzweifel/git-auto-commit-action#339] [Ideas]
I only want to commit changes (made by code formatters) to existing files. However, in order to format the files, I download other GitHub actions into the working directory. They get committed as submodules by this action.

My workaround is using `add_options: '-n'`. However, this feels like a hack/workaround, and I'd appreciate a `skip_add` option similar to `skip_fetch` and `skip_checkout`.

**Comment by stefanzweifel** [comment]
In such cases I would suggest adding your submodule to `.gitignore` so it won't be added by the action.

Adding a `skip_add`-option doesn't make sense to me; maybe I just don't see the issue as clearly as you do as I don't have an explicit example. 
If the `git-add`-step is skipped, the core feature of the action does not work. 🤔

**Reply by janopae** [reply]
You're right. The behaviour I actually wanted was the equivalent of providing no pathspec argument to `add` or using the [`-u` option](https://git-scm.com/docs/git-add#Documentation/git-add.txt--u). That seems to work perfectly fine :)

(at least the `-u` option – haven't tried the empty pathspec)
